### DISCUSSION TITLE
beneficiary.name

### DISCUSSION THREAD

**Discussion by romulan666** [xseman/bysquare#21] [Q&A] [answered]
Ahoj,
vedel by si mi prosim poradit ako do example.json suboru pridam moznost BeneficiaryName? Skusal som rozne moznosti ale nic z toho mi nefunguje. Inak super praca dakujem.

**Comment by xseman** [comment, marked as answer]
https://github.com/xseman/bysquare/blob/master/src/types.ts#L156-L175

Tento interface by ti mohol pomôcť:

```ts
export type Beneficiary = {
	/**
	 * Maximálna dĺžka 70
	 *
	 * Rozšírenie o meno príjemcu
	 */
	name?: string;
	/**
	 * Maximálna dĺžka 70
	 *
	 * Rozšírenie o adresu príjemcu
	 */
	street?: string;
	/**
	 * Maximálna dĺžka 70
	 *
	 * Rozšírenie o adresu príjemcu (druhý riadok)
	 */
	city?: string;
};
```

`example.json` by mohol vyzerať takto:

```json
{
	"invoiceId": "random-id",
	"payments": [
		{
			"type": 1,
			"amount": 100.0,
			"bankAccounts": [{ "iban": "SK9611000000002918599669" }],
			"currencyCode": "EUR",
			"variableSymbol": "123",
			"beneficiary": {
				"name": "tvoje meno"
			}
		}
	]
}
```

Najideálnejšie je si to vyskladať v TypeScripte, ten ti pekne hintuje možnosti.

**Reply by romulan666** [reply]
Dakujem velmi pekne. Pomohlo.
